Donation-platform/src/main/java/com/google/DbCalls.java

```diff
@@ -110,17 +97,144 @@ public void addUser(String userName, String email,
+  public Tag getTagByName(String name) throws Exception{
+    Query query =
+    new Query("Tag")
+        .setFilter(new FilterPredicate("name", FilterOperator.EQUAL, name));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Tag tagResult = setTagClass(result);
+    return tagResult;
+  }
+  //Function that returns user from user name.
+  public Users getUserByUserName(String userName) throws Exception{
+    Query query =
+    new Query("Users")
+        .setFilter(new FilterPredicate("userName", FilterOperator.EQUAL, userName));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Users userResult = setUsersClass(result);
+    return userResult;
+  }
+  // Function that returns user from email.
+  public Users getUserByEmail(String email) throws Exception{
+    Query query =
+    new Query("Users")
+        .setFilter(new FilterPredicate("email", FilterOperator.EQUAL, email));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Users userResult = setUsersClass(result);
+    return userResult;
+  }
+
+  // Function turns entity into class.
+  public Charity setCharityClass(Entity entity){
+    Key id = (Key) entity.getKey();
+    String name = (String) entity.getProperty("name");
+    String link = (String) entity.getProperty("link");
+    List<Key> categories = (List<Key>) entity.getProperty("categories");
+    String description = (String) entity.getProperty("description");
+    Double trendingScore = (Double) entity.getProperty("trendingScore");
+    // Converts data to charity object.
+    Charity charityData = new Charity(id, name, link, categories, description, trendingScore);
+    return charityData;
+  }
+  // Function turns entity into class.
+  public Tag setTagClass(Entity entity){
+    Key id = (Key) entity.getKey();
+    String name = (String) entity.getProperty("name");
+    Double trendingScore = (Double) entity.getProperty("trendingScore");
+    // Converts data to tag object.
+    Tag tagData = new Tag(id, name, trendingScore);
+    return tagData;
+  }
+  // Function turns entity into class.
+  public Users setUsersClass(Entity entity){
+    Key id = (Key) entity.getKey();
+    String userName = (String) entity.getProperty("userName");
+    String email = (String) entity.getProperty("email");
+    List<Key>  userInterests = (List<Key> ) entity.getProperty("userInterests");
+    List<Key>  charitiesDonatedTo = (List<Key> ) entity.getProperty("charitiesDonatedTo");
+    // Converts data to users object.
+    Users userData = new Users(id, userName, email, userInterests, charitiesDonatedTo);
+    return userData;
+  }
+  // Function takes modified class and updates database.
+  public void updateCharity(Charity charity) throws Exception{
+    Key id = charity.getId();
     Entity charityEntity = datastore.get(id);
-    charityEntity.setProperty("trendingScore", trendingScore);
```

Comment:
declare constants for all these strings at the top of file and use it throughout.

Reply:
resolved.